cdk-project/lib/images/codebuild-image/python/src/notebooks/parse.py

```diff
@@ -225,3 +244,46 @@ def skip(notebook):
     elif any([str(directory) in SKIP_LIST for directory in directories]):
         return True
     return False
+
+def uses_docker(notebook):
+    """Check whether the notebook should be skipped because it uses docker based functionality.
+
+    Args:
+        notebook (Path): The notebook to check.
+
+    Returns:
+        bool: True if the notebook uses docker.
+
+    """
+    return contains_code(
+        notebook, ["docker ", "docker-compose ", "Docker "]
+    )
+
+def local_mode_mandatory(notebook):
+    """Check whether the notebook should be skipped because runs only in Local Mode
+
+    Args:
+        notebook (Path): The notebook to check.
+
+    Returns:
+        bool: True if local mode is mandatory.
+
+    """
+    return (
+            (str(notebook) not in LOCAL_MODE_OPTIONAL_LIST) and
+            (contains_code(notebook, ['instance_type = "local"', 'instance_type="local"']))
```

Comment:
please use regex here to check <space> after "="

Reply:
Updated